pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -66,22 +65,23 @@ type PCCloud struct {
 	// in the environment per current Photon Controller deployment methodology.
 	projID string
 	cloudprovider.Zone
+	photonClient *photon.Client
```

Comment:
Here I tried to follow the same pattern that is used in photon controller CLI. It's like every time the user calls a photon controller CLI command, a new photon client is created. And this client won't be accessed by multiple threads. It will only be used in the current operation. Hopefully this means we are safe (at least for now).